### PR TITLE
Organise config data and dependencies in gradle

### PR DESCRIPTION
**Pull request by ezisazis**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -21,5 +21,9 @@ android {
 }
 
 dependencies {
-    compile 'com.android.support:support-v4:25.3.1'
+    compile "com.android.support:support-v4:$rootProject.supportLibraryVersion"
+
+    // UI Related
+    compile "com.jakewharton:butterknife:$rootProject.butterKnifeVersion"
+    annotationProcessor "com.jakewharton:butterknife-compiler:$rootProject.butterKnifeVersion"
```
Comment (sionleroux) [inline app/build.gradle]: I bet this line is the one conflicting with master, please rebase and fix it :wink:

File: build.gradle [hunk 2]
```diff
@@ -13,3 +13,30 @@ allprojects {
...
+
+    // Support
+    supportLibraryVersion = "25.3.1"
+
+    // UI
+    butterKnifeVersion = "8.5.1"
+}
+
+def computeVersionCode() {
+    return configuration.versionCodeMajor * 100 + configuration.versionCodeMinor * 10 + configuration.versionCodePatch
```
Comment (sionleroux) [inline build.gradle:37]: Can I see the result of this statement somewhere?

File: build.gradle [hunk 3]
```diff
@@ -13,3 +13,30 @@ allprojects {
...
+}
+
+def computeVersionName() {
+    return "$configuration.versionCodeMajor.$configuration.versionCodeMinor.$configuration.versionCodePatch"
+}
```
Comment (sionleroux) [inline build.gradle]: For human-editable files please always include a newline at the end of the file, you should be able to set your editor to do this for you, in fact I just added settings for this in master with 2eef933f4a9a9b9cde946e146876f5fb38ae41d3, so a rebase and opening-and-closing-in-IDE should help fix this too.

**Review by sionleroux — CHANGES_REQUESTED** [review]
Please rebase to fix really tiny nit picks, apart from that it looks cool! :purple_heart:

**Comment by sinisterstuf** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please limit commit subject line to 50 characters.<br />
e81ee70c49c6453f157a7fa1bd394fb58b68f13f</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: build.gradle (hunk 2 above)
Comment (ezisazis) [reply build.gradle:37]: When you run the code you can log it 🥇

File: build.gradle (hunk 3 above)
Comment (ezisazis) [reply build.gradle]: done!

**Comment by ezisazis** [comment]
@sinisterstuf everything is fine now

**Review by sionleroux — APPROVED** [review]
Awesome!

**Comment by sionleroux** [comment]
@ezisazis I added the missing translations that were breaking the build in 003dc36 so it's green again on master, can you rebase this branch to that commit please, so that this build can pass too?

**Comment by ezisazis** [comment]
@sinisterstuf done!

**Comment by sinisterstuf** [comment]
Thanks for organising stuff around! Can you explain why it's better to have these values in build.gradle than in app/build.gradle?

Also why it's better to have

```
x = 1
y = 0
z = 0
version = x.y.z
```

instead of 

```
version = 1.0.0
```

? I mean it's definitely _cool_, but why is it good?

**Pull request merged**